### PR TITLE
Build mraa upm libraries

### PR DESCRIPTION
**Pull request by IvanFarkas**
Added **Build MRAA and UPM Libraries** guide

### CODE REVIEW

**Comment by Mani-Sadhasivam** [comment]
@ndechesne From what I heard from @fboudra during connect is, debian MRAA package itself is very old and I don't think it is being maintained.

**Comment by IvanFarkas** [comment]
There are substantial changes that my future guides, blogs rely on.

- NodeJS, NPM, SWIG have to be the latest.
- Documents need to be generated.
- C# SWIG interface will be added to UPM.

Do you want me to update the existing one to reflect these must-have features?

**Comment by sdrobertw** [comment]
@ndechesne @IvanFarkas @Mani-Sadhasivam 

I am fine with merging this guide; however, is it not a replica of what you already have @Mani-Sadhasivam ?

The generic guides sections are being shuffled and re-organized right now, but eventually it should all be in order.

If we merge this as a DB820c specific guide, I am not sure if that is the right direction. Even though, merging it here will not do any damage.

**Comment by Mani-Sadhasivam** [comment]
If we merge this guide, then it will add another level of fragmentation (we already have enough of these).

@IvanFarkas Can you please add UPM doc to /consumer/guides/ as I suggested earlier. It would barely be a copy paste work. Please follow /consumer/guides/mraa structure.

Many thanks for your work :)

**Comment by IvanFarkas** [comment]
Sure.
My guides must use the latest nodejs.
It requires a patch for SWIG 4.0.
Do we have a separate guide or update the current one?

**Comment by Mani-Sadhasivam** [comment]
There is no guide AFAIK. But I'm aware of the issue with SWIG and Node.js. So, please go and update the current one and mention that it is only needed for the more recent version of Node.js.

**Comment by IvanFarkas** [comment]
That’s a good plan

File: consumer/guides/innstall_upm.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+---
```
Comment (Mani-Sadhasivam) [inline consumer/guides/innstall_upm.md]: Can you move this file under /consumer/guides/upm/ ? Also the file name should be install_upm.md

File: consumer/guides/mraa/install.md [hunk 2]
```diff
@@ -2,12 +2,22 @@
...
+
 ---
 
 # Installation guide for MRAA library on 96Boards CE
 
 Following instructions provides information about installing the MRAA library as a package and building from source.
 
+## MRAA Overview
+**[MRAA](https://iotdk.intel.com/docs/master/mraa/)** (libmraa) is a C/C++ low level skeleton library for communication on GNU/Linux platforms.
+It binds to Python, Javascript and Java to interface with the I/O on ARM platforms such as the DragonBoard 820c.
```
Comment (Mani-Sadhasivam) [inline consumer/guides/mraa/install.md]: Please don't specify any particular board name here. This guide is meant for all of the 96Boards

File: consumer/guides/mraa/install.md [hunk 3]
```diff
@@ -2,12 +2,22 @@
...
+It has a structured and sane API where port names/numbering matches the board that you are on.
+Use of libmraa does not tie you to specific hardware with board detection done at runtime you can create portable code that will work across the supported platforms.
+MRAA make it easier for developers and sensor manufacturers to map their sensors & actuators on top of supported hardware and to allow control of low level communication protocol by high level languages & constructs.
+
+Recent addition to the MRAA library by **[Manivannan Sadhasivam](https://twitter.com/mani_sadhasivam)** and **[Ivan Farkas](https://twitter.com/ifarkas)** enabled and extended MRAA for the DragonBoard 820c.
```
Comment (Mani-Sadhasivam) [inline consumer/guides/mraa/install.md]: No need of this information here. The author information can be found in the MRAA commit history.

File: consumer/guides/mraa/install.md [hunk 4]
```diff
@@ -21,40 +31,262 @@ A few useful commands will help us make sure everything on the board is current:
...
 
-Following instructions specifies installing MRAA library from source.
+Following sections specifies installing MRAA library from source. We build and install dependencies first.
+If you want to use legacy nodejs you can skip nodejs and SWIG sections and just get them from their packages.
+If you want to be current, than do alls teps as outlined below.
+
+**Install extra tool packages**
+
+```shell
+sudo apt-get install -y arduino-mk arduino git build-essential cmake bison doxygen graphviz python-sphinx python3-sphinx libudev-dev libgtest-dev libjson-c-dev autoconf libtool python-dev pkg-config libpcre3-dev
```
Comment (Mani-Sadhasivam) [inline consumer/guides/mraa/install.md]: Aren't we installing too many packages here? Why do we need arduino packages?

File: consumer/guides/mraa/install.md [hunk 5]
```diff
@@ -21,40 +31,262 @@ A few useful commands will help us make sure everything on the board is current:
...
+sudo apt-get clean
+sudo apt autoremove -y
+```
+
+**We install all repos into the git directory to stay organized**
```
Comment (Mani-Sadhasivam) [inline consumer/guides/mraa/install.md]: These informations are very useful to someone who likes to use JAVA bindings but not for everyone. Can you please create two sections? One for a newcomer and another for advanced user? If you don't have time, no issues, I'll do it once this guide gets merged.

File: consumer/guides/mraa/install.md [hunk 6]
```diff
@@ -21,40 +31,262 @@ A few useful commands will help us make sure everything on the board is current:
...
+
+**Output**
+
+```shell
+SWIG Version 4.0.0
+Compiled with g++ [aarch64-unknown-linux-gnu]
+Configured options: +pcre
+```
+
+## Install and build Google test library
```
Comment (Mani-Sadhasivam) [inline consumer/guides/mraa/install.md]: Why do we need this?

File: consumer/guides/mraa/install.md [hunk 7]
```diff
@@ -21,40 +31,262 @@ A few useful commands will help us make sure everything on the board is current:
...
+cd build
+cmake \
+ -DCMAKE_INSTALL_PREFIX:PATH=/usr/local \
+ -DCMAKE_BUILD_TYPE=DEBUG \
+ -DCMAKE_C_COMPILER=/usr/bin/clang-6.0 \
+ -DCMAKE_CXX_COMPILER=/usr/bin/clang++-6.0 \
+ -DBUILDDOC=ON \
+ -DBUILDSWIGJAVA=ON \
+ -DBUILDSWIGNODE=ON \
+ -DBUILDSWIGPYTHON=OFF \
```
Comment (Mani-Sadhasivam) [inline consumer/guides/mraa/install.md]: Why? Python is one of the most used interface. We should enable them.

**Comment by sdrobertw** [comment]
@Mani-Sadhasivam 

Should this PR be merged or closed?

**Comment by IvanFarkas** [comment]
Merge, please

**Comment by Mani-Sadhasivam** [comment]
@IvanFarkas You still haven't updated the PR as per my review comments! Please do it and we will merge it. Many thanks once again.

**Comment by sdrobertw** [comment]
@Mani-Sadhasivam seems I will be closing this PR.

Please advise, if no fix, it will be closed by EoW.

Thanks.

**Comment by IvanFarkas** [comment]
OK

**Comment by sdrobertw** [comment]
@IvanFarkas Thanks!

**Pull request closed without merging**